packages/react/src/auto/polaris/inputs/PolarisAutoEnumInput.tsx

```diff
@@ -0,0 +1,149 @@
+
+  const selectedTagsElement = useMemo(
+    () =>
+      allowMultiple && selectedOptions.length > 0 ? (
+        <InlineStack gap="150">
+          {selectedOptions.map((tag) => (
+            <Tag key={`option-${tag}`} onRemove={() => onSelectionChange(tag)}>
+              {tag}
+            </Tag>
+          ))}
+        </InlineStack>
+      ) : null,
+    [allowMultiple, onSelectionChange, selectedOptions]
+  );
+
+  const optionItemElement = useMemo(() => {
+    if (allOptions.length === 0) return null;
+
+    const formatOptionText = (option: string) => {
+      const trimValue = searchValue.trim().toLocaleLowerCase();
+      const matchIndex = option.toLocaleLowerCase().indexOf(trimValue);
+
+      if (!searchValue || matchIndex === -1) return option;
+
+      const start = option.slice(0, matchIndex);
+      const highlight = option.slice(matchIndex, matchIndex + trimValue.length);
+      const end = option.slice(matchIndex + trimValue.length, option.length);
+
+      return (
+        <p>
+          {start}
+          <Text fontWeight="bold" as="span">
+            {highlight}
+          </Text>
+          {end}
+        </p>
+      );
+    };
+
+    return filteredOptions.map((option) => {
+      return (
+        <Listbox.Option key={option} value={option} selected={selectedOptions.includes(option)} accessibilityLabel={option}>
+          <Listbox.TextOption selected={selectedOptions.includes(option)}>{formatOptionText(option)}</Listbox.TextOption>
+        </Listbox.Option>
+      );
+    });
+  }, [allOptions.length, filteredOptions, searchValue, selectedOptions]);
+
+  const addExtraOptionElement = useMemo(() => {
+    if (!allowOther) return null;
+
+    const noResults = searchValue && !allOptions.includes(searchValue);
+    return noResults ? <Listbox.Action value={searchValue}>{`Add "${searchValue}"`}</Listbox.Action> : null;
+  }, [allOptions, allowOther, searchValue]);
+
+  const emptyStateElement = useMemo(() => {
+    if (allowOther || (optionItemElement && optionItemElement.length > 0) || !searchValue) return null;
+    return <EmptySearchResult title="" description={`No  found matching "${searchValue}"`} />;
+  }, [allowOther, optionItemElement, searchValue]);
+
+  useEffect(() => {
+    if (allowMultiple) {
+      setSearchValue("");
+      return;
+    }
+
+    if (selectedOptions.length > 0) {
+      // Set the search value to show the selected option in the input field
+      setSearchValue(selectedOptions[0]);
+    } else {
+      setSearchValue("");
+    }
+  }, [allowMultiple, selectedOptions, setSearchValue]);
+
+  const listBoxElement = useMemo(
+    () =>
+      optionItemElement || addExtraOptionElement || emptyStateElement ? (
+        <Listbox
+          autoSelection={AutoSelection.None}
+          onSelect={(selected) => {
+            onSelectionChange(selected);
+            console.log("selectedOptions:", selectedOptions);
+            if (allowMultiple) {
```

Comment:
As per previous comment, if you extracted a 'clearSearch' function you could call it here